### ISSUE TITLE
Parser fails to parse empty strings

### ISSUE THREAD

**Issue by wwilfinger** [metrico/qryn#242]
Hello!

Grafana's "Explore" defaults with the query builder turned on. After you select a label filter Grafana builds a query like `{container="example"} |= `` ` Qryn fails to parse these queries, but Loki runs them just fine. From guessing and checking it seems like it's the empty string that it doesn't like.

```
#!/usr/bin/env node
const { transpile } = require('qryn/parser/transpiler')  # 2.1.43

# parses
console.log(transpile({
  query: "{container=\"example\"} |= \`example log\`"
}).query)

# fails
console.log(transpile({
  query: "{container=\"example\"} |= \`\`"
}).query)
```

The error is

```
Couldn't execute script:
Expected [ 'EOF', or 'WSP' ] @1:22 encountered ' |='.
{container="example"} |= ``
/home/.../node_modules/qryn/parser/transpiler.js:90
  const token = expression.rootToken
                           ^

TypeError: Cannot read properties of null (reading 'rootToken')
    at module.exports.transpile (/home/.../node_modules/qryn/parser/transpiler.js:90:28)
```

For what it's worth, the animation with Grafana in the Qryn docs [here](https://qryn.metrico.in/#/getting-started) does show results popping up with the ` |= `` ` in the LogQL query.

**Comment by lmangani** [comment]
Hello @wwilfinger 

This doesn't seem to be a qryn but, but rather a Grafana one. We observed this bug on broken Grafana versions (9.1.x)
Could you kindly specify the version of both qryn and grafana in use?

**Comment by lmangani** [comment]
Interesting. Does it work when you remove the empty match condition?

**Comment by lmangani** [comment]
@wwilfinger fixed in qryn@2.1.44 by @akvlad ⚡ 

Thanks for your report and support! Please let us know if this validates in your setup.

**Comment by akvlad** [comment]
not yet. Working on it.

**Comment by lmangani** [comment]
Looks fixed in qryn@2.1.45 and all tests passing

**Comment by wwilfinger** [comment]
Yep looks like v2.1.45 did the trick. Thanks so much for all your help!
